generators/react/templates/src/main/webapp/app/shared/util/entity-utils.ts.ejs

```diff
@@ -60,3 +60,13 @@ export const overridePaginationStateWithQueryParams = (paginationBaseState: IPag
   }
   return sortedPaginationState;
 };
+
+export const prepareValuesToSave = (values, entity) => {
+  const result = { ...values };
+  for (const property in values) {
+    if (typeof entity[property] === 'number' && typeof values[property] === 'string') {
```

Comment:
When 'isNew' is true, `entity == {}`, making it impossible to retrieve type information and thus perform type conversion. Solving this issue using a generic utility function like 'prepareValuesToSave' is challenging. I believe it needs to be addressed through code generation, similar to how date type processing is handled.

https://github.com/jhipster/generator-jhipster/blob/72de10a384c263a7af972742669e043de6021282/generators/react/templates/src/main/webapp/app/entities/_entityFolder_/_entityFile_-update.tsx.ejs#L155-L157

Reply:
@hide212131 yes you're right, I updated the PR